### ISSUE TITLE
Separate Post-Processing and Classification Logic

### ISSUE THREAD

**Issue by dni138** [mozilla-ai/any-guardrail#23]
**Why would we implement this?**

By separating the logic of post-processing and classification out of `safety_review` we are affording the package to allow for API support. Essentially, the APIs of guardrails may produce logits or text output, etc., but have their own system prompts and calling logic, separate from the classification portion of the `safety_review`.

**How to implement?**

Go through the code and find places where we are overloading `safety_review` and then create hidden functions that implement the classification and post-processing logic separately.
